sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -407,6 +408,14 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
       Row(1, Row(Row(1, "a"), Row(1, "a"))) :: Row(2, Row(Row(2, "bc"), Row(2, "bc"))) :: Nil)
   }
 
+  test("SPARK-24598: sum throws exception instead of silently overflow") {
```

Comment:
If we decide to defer the change of default behavior to a separate PR then I think we'd need to do a `withSQLConf` in this test to opt in to the new behavior. As long as we're doing that, maybe we should test both sides of the feature flag, e.g. test that we _do_ continue to silently overflow when the flag has the old value set.